internal/debug/api.go

```diff
@@ -237,6 +237,14 @@ func (*HandlerT) SetGCPercent(v int) int {
 	return debug.SetGCPercent(v)
 }
 
+// SetMemoryLimit sets the GOMEMLIMIT for the process. It returns the previous limit.
+// Note:
+//   - Geth also allocates memory off-heap, particularly for fastCache and Pebble, which can be non-trivial (a few gigabytes by default).
+//   - Setting the limit too low will cause Geth to become unresponsive.
```

Comment:
Could you at least add a comment stating that the input limit is specified in bytes? I’m not sure whether we should sanitize excessively low values, as it’s unclear where the threshold should be. 

But I can foresee that people might pass the values as the megabytes or so, resulting in an unexpected behavior.

I would also prefer to print out some logs so that users know what's the threshold being set

Reply:
Sorry for the late reply. I will submit as soon

Reply:
> I’m not sure whether we should sanitize excessively low values, as it’s unclear where the threshold should be.

The minimum value is not easy to decide, I added a comment if the setting is too small